src/wallet/rpcwallet.cpp

```diff
@@ -3056,6 +3060,15 @@ UniValue listunspent(const JSONRPCRequest& request)
         entry.pushKV("scriptPubKey", HexStr(scriptPubKey.begin(), scriptPubKey.end()));
         entry.pushKV("amount", ValueFromAmount(out.tx->tx->vout[out.i].nValue));
         entry.pushKV("confirmations", out.nDepth);
+        if (out.nDepth) {
+            LOCK(mempool.cs);
+            const auto it = mempool.mapTx.find(out.tx->GetHash());
+            if (it != mempool.mapTx.end()) {
+                entry.pushKV("ancestorcount", it->GetCountWithAncestors());
+                entry.pushKV("ancestorsize", it->GetSizeWithAncestors());
+                entry.pushKV("ancestorfees", it->GetModFeesWithAncestors());
+            }
+        }
```

Comment:
Should this also add the no ancestor case?
```
{
    ...
    "ancestorcount": 0,
    "ancestorsize": 0,
    "ancestorfees": 0,
    ....
}
```
If the aim is to avoid adding unnecessary size to responses, then the help message above might mention these fields will only print if ancestors exist.

Reply:
ancestor* fields always include the transaction itself